test/e2e/traceflow_test.go

```diff
@@ -1159,6 +1159,27 @@ func testTraceflowInterNode(t *testing.T, data *TestData) {
 	agnhostIP, err := data.podWaitForIPs(defaultTimeout, agnhostPodName, data.testNamespace)
 	require.NoError(t, err)
 
+	node2TunnelDstIPv4, node2TunnelDstIPv6 := "", ""
+	encapMode, err := data.GetEncapMode()
+	require.NoError(t, err)
+
+	switch encapMode {
+	case config.TrafficEncapModeEncap:
+		node2TunnelDstIPv4 = clusterInfo.nodes[nodeIdx1].ipv4Addr
+		node2TunnelDstIPv6 = clusterInfo.nodes[nodeIdx1].ipv6Addr
+	case config.TrafficEncapModeHybrid:
+		// check if node1 and node2 are in same subnet.
+		node2IP := net.ParseIP(nodeIP(nodeIdx1))
+		cmd := getPingCommand(pingCount, 0, "", &node2IP)
+		stdout, _, err := data.RunCommandFromPod(data.testNamespace, node1Pods[0], agnhostContainerName, cmd)
+		require.NoError(t, err)
+		if !(strings.Contains(stdout, "ttl=63") || strings.Contains(stdout, "ttl=127")) {
```

Comment:
This largely increases the test's dependency on testbeds: it requires firewalls opened for ICMP and the initial TTL value varies on different systems.
You could just get the network mask of nodeIP1 and checks if the subnet contains nodeIP2.
However, this is not something having to be in the fix. Please let me know if you could make the change quickly, or let's just fix the test first without adding more validation.

Reply:
@tnqn Is it possible to get the Node's network mask in e2e script without run shell command on node1?
If not, I recommend to use `ip route get nodeIP2 | grep via` on node1 instead.

Reply:
I suggested netmask to avoid adding another dependency on iproute2 tool as they are not always available.

Reply:
getting the network mask in e2e script will make the test suite harder to run, as we will have to run the test suite via the script, instead of just `go test` which we have documented somewhere.

Reply:
Shall we skip this validation in Hybrid mode e2e and add a TODO here?
I think the best way is to get the routed/tunneled remote Node via an official method, e.g. AgentInfo CRD.

Reply:
Skipping the tunnelIP validation in the PR works for me. It could focus on the bug fix.

Reply:
skipped validation for hybrid mode.